src/hooks/useSyncChainQuery.ts

```diff
@@ -21,8 +21,18 @@ function getParsedChainId(parsedQs?: ParsedQs) {
 }
 
 export default function useSyncChainQuery() {
-  const { chainId, isActive } = useWeb3React()
+  const { chainId, isActive, account } = useWeb3React()
   const parsedQs = useParsedQueryString()
+  const chainIdRef = useRef(chainId)
+  const accountRef = useRef(account)
+
+  useEffect(() => {
+    // Update chainIdRef when the account is retrieved from Web3React
+    if (account && account !== accountRef.current) {
+      chainIdRef.current = chainId
```

Comment:
Why can't this be either outside of a `useEffect` or grouped in with the _other_ `useEffect`?

Reply:
We need to monitor changes to account and chainId for this change but only these values which is a subset of the larger useEffect. This keeps this function from unnecessarily running if there's a state change to one of the other dependencies in the larger useEffect